core/src/main/java/com/exadel/kaliada/core/services/impl/HarvardNewsParserImpl.java

```diff
@@ -0,0 +1,127 @@
+package com.exadel.kaliada.core.services.impl;
+
+import com.day.cq.dam.api.Asset;
+import com.day.cq.dam.api.AssetManager;
+import com.exadel.kaliada.core.services.HarvardNewsParser;
+import com.exadel.kaliada.core.utils.ResourceResolverUtil;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.sling.api.resource.LoginException;
+import org.apache.sling.api.resource.ResourceResolver;
+import org.apache.sling.api.resource.ResourceResolverFactory;
+import org.jsoup.Jsoup;
+import org.jsoup.nodes.Document;
+import org.jsoup.nodes.Element;
+import org.jsoup.select.Elements;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+import java.io.BufferedInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URL;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Class to load news from https://news.harvard.edu/gazette/ and parse them
+ * @author akaliada
+ */
+
+@Slf4j
+@Component(service = HarvardNewsParserImpl.class)
+public class HarvardNewsParserImpl implements HarvardNewsParser {
+
+    private static final String TITLE_PATH = "/jcr:content/root/container/title?jcr:title";
+    private static final String TEXT_PATH = "/jcr:content/root/container/text?text";
+    private static final String ASSET_PATH = "/content/dam/aemdemo";
+    private static final String DEFAULT_IMAGE_PATH = "/content/dam/aemdemo/harvard-university-free-online-courses.jpg";
+    public static final String IMAGE_PATH = "/jcr:content/root/container/image?fileReference";
+
+    @Reference
+    private ResourceResolverFactory resourceResolverFactory;
+
+    /**
+     * load and parse all news from the main web-site
+     * @return Map of pairs news Id and news link
+     */
+
+    @Override
+    public Map<String, String> getAllNews() {
+        try {
+            Map<String, String> newsIdToLink = new HashMap<>();
+            Document document = Jsoup.connect("https://news.harvard.edu/gazette/")
+                    .userAgent("Chrome/4.0.249.0 Safari/532.5")
+                    .referrer("https://www.google.com")
+                    .get();
+            Elements news = document.select("article[id]");
+            for (Element element : news) {
+                String id = element.attr("id");
```

Comment:
Please use constants extensively. As for some local tradition, we rarely use "inline literals", though some debate does periodically arise